### PR TITLE
fix: trailing comma not allowed in ruleset

### PR DESCRIPTION
**Pull request by eduardomourar**
*Issue #, if available:*
#633 

*Description of changes:*

This will generate a valid TypeScript file for `src/endpoint/ruleset.ts`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddDefaultsModeDependency.java [hunk 1]
```diff
@@ -37,10 +37,10 @@ public void addConfigInterfaceFields(
...
         writer.addDependency(TypeScriptDependency.AWS_SDK_UTIL_DEFAULTS_MODE_NODE);
-        writer.addImport("DefaultsMode", "DefaultsMode", TypeScriptDependency.AWS_SMITHY_CLIENT.packageName);
-        writer.addImport("Provider", "Provider", TypeScriptDependency.AWS_SDK_TYPES.packageName);
-        writer.writeDocs("The {@link DefaultsMode} that will be used to determine how certain default configuration "
+        writer.addImport("DefaultsMode", "__DefaultsMode", TypeScriptDependency.AWS_SMITHY_CLIENT.packageName);
+        writer.addImport("Provider", "__Provider", TypeScriptDependency.AWS_SDK_TYPES.packageName);
+        writer.writeDocs("The {@link __DefaultsMode} that will be used to determine how certain default configuration "
                 + "options are resolved in the SDK.");
-        writer.write("defaultsMode?: DefaultsMode | Provider<DefaultsMode>;\n");
+        writer.write("defaultsMode?: __DefaultsMode | __Provider<_DefaultsMode>;\n");
```
Comment (gosar) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddDefaultsModeDependency.java:44]: These changes seem unrelated. Do you need them?

**Review by gosar — COMMENTED** [review]
I added this [comment](https://github.com/awslabs/smithy-typescript/issues/633#issuecomment-1398017516) to the issue. But looking at the code and your diff here, I can see that it would generate `,` and is likely a prettify step that's fixing it in AWS SDK. To showcase that more directly, and for adding some missing testing here, can we add a test  that would be broken before your change and fixed with this.

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddDefaultsModeDependency.java (hunk 1 above)
Comment (eduardomourar) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddDefaultsModeDependency.java:44]: by following the expected convention, we can avoid things like this: https://github.com/aws/aws-sdk-js-v3/blob/3844108f9adcd8282f66189427bad8cb309d2746/clients/client-accessanalyzer/src/AccessAnalyzerClient.ts#L45-L46

Comment (eduardomourar) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddDefaultsModeDependency.java:44]: Created this PR for it: https://github.com/awslabs/smithy-typescript/pull/670

**Review by kuhe — APPROVED** [review]

**Review by gosar — APPROVED** [review]

**Comment by eduardomourar** [comment]
@gosar, create PR #674 to include a test case called `containsTrailingSemicolon` where now it has been fixed.

**Comment by gosar** [comment]
> @gosar, create PR #674 to include a test case called `containsTrailingSemicolon` where now it has been fixed.

Thanks @eduardomourar! I merged your PR as I didn't want to hold up the fix and was going to work on adding tests later. Thanks for adding them :)
